workers/unity/Assets/Playground/Scripts/Player/MobilePlayerInputSystem.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using Unity.Entities;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Playground
+{
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    [UpdateBefore(typeof(LocalPlayerInputSync))]
+    public class MobilePlayerInputSystem : ComponentSystem
+    {
+        private struct InputData
+        {
+            public readonly int Length;
+            public ComponentDataArray<LocalInput> PlayerInput;
+        }
+
+        [Inject] private InputData inputData;
+
+        private static GameObject controllerJoystick;
+        private static GameObject cameraControllerJoystick;
+        private static VirtualJoystick movementJoystick;
+        private static VirtualJoystick cameraJoystick;
+        private static HashSet<int> oldTouchSet;
+        private static HashSet<int> newTouchSet;
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            try
+            {
+                controllerJoystick = GameObject.FindGameObjectWithTag("MovementJoystick");
+                movementJoystick = controllerJoystick.GetComponent<VirtualJoystick>();
+            }
+            catch (NullReferenceException)
+            {
+                WorkerRegistry.GetWorkerForWorld(World)
+                    .View.LogDispatcher.HandleLog(LogType.Error,
+                        new LogEvent("Could not find movement virtual stick. Movement is now disabled on mobile"));
+            }
+
+            try
+            {
+                cameraControllerJoystick = GameObject.FindGameObjectWithTag("CameraJoystick");
+                cameraJoystick = cameraControllerJoystick.GetComponent<VirtualJoystick>();
+            }
+            catch (NullReferenceException)
+            {
+                WorkerRegistry.GetWorkerForWorld(World).View.LogDispatcher.HandleLog(LogType.Error,
+                    new LogEvent("Could not find virtual camera joystick. Camera movement is now disabled on mobile"));
+            }
+
+            oldTouchSet = new HashSet<int>();
+            newTouchSet = new HashSet<int>();
```

Comment:
this can be moved outside of `OnCreateManager` to directly initialize the fields